### DISCUSSION TITLE
[❔] Where to put documentation used during planning and/or development of the module?

### DISCUSSION THREAD

**Discussion by claudio-salvio** [gaelcolas/Sampler#496] [Q&A]
Hi Samplers ! 👋

During the development of a software package (in this case a module) it is common to write documentation used in the process.

This documentation I am referring to is not directly related to the functionality that the module will have once released (it is not the module's help) but rather to aspects more related to its development.
For example: use cases , class diagrams, specifications, ideas, etc.

Any suggestions on where to place these types of files within the folder structure of a Sampler-based project?
Do you recommend creating a specific folder for this purpose?

Best regards,
Claudio Salvio

**Comment by johlju** [comment]
Maybe a `docs/development` folder and CONTRIBUTING.md is structuring and pointing to different markdown pages 🤔 Folder `docs` and README.md might be used by some projects for the documentation of the functionality. 

In the SqlServerDsc module project everything is put inside CONTRIBUTING.md which is not optimal either, it becomes a bit unstructured and not easy to find.

There is another option if using GitHub, by using WikiSource and publish do the repository wiki it possible to make a section in the side bar for example "Development", see example for "Usage" here: https://github.com/dsccommunity/DscResource.AnalyzerRules/wiki
By adding the following to the top of the markdown page in WikiSource it is possible to make sections in the side bar:

```markdown
---
Category: Usage
---
```

**Reply by claudio-salvio** [reply]
Hi @johlju 
Thanks for your suggestions !
